packages/dom/README.md

```diff
@@ -204,7 +204,7 @@ and has a valueAsNumber
 
 _Parameters_
 
--   _element_ `HTMLElement`: The HTML element.
+-   _element_ `Partial<HTMLInputElement>`: The HTML element.
```

Comment:
Pretty new to this. Why is `Partial` needed?

Reply:
`Partial` is needed because the `Element` itself could be any HTML element that we're testing to check to see if it's an input, textarea or `contentEditable` Element. I wasn't sure how else to express that the `Input` part is optional and I think this is what @sirreal suggested as a potential solution to this problem (though I may be mis remembering, I can't find the original comment where that was suggested.

Reply:
I believe this was introduced because we have some type that `HTMLInputElement` extends, but we don't know that it's an `HTMLInputElement`.
`Partial<HTMLInputElement>` doesn't seem ideal to me, it's typically used when we have an interface with optional properties. This usage is really that we have an `Element` (I believe, could be another type) and want to use it safely if it satisfies `HTMLInputElement`.

In my opinion, a better approach would be to accept the element type we expect to pass in (`Element`?), and use a type guard to narrow our element to our desired type. I implemented a type guard like this in #29220 to handle this type of case:

https://github.com/WordPress/gutenberg/blob/29601bd2a0427037dde074a376d4b23e2ae61460/packages/dom/src/dom.js#L83-L89

@sarayourfriend I believe you're thinking of this comment: https://github.com/WordPress/gutenberg/pull/30030#discussion_r598714524

Reply:
Right, maybe just `HTMLElement`? We should allow any element to be checked with this function.

Reply:
Aha, yes, I did try to create a type guard but got stuck on `contentEditable`... I'll revisit this though and see if I can make it work with a type guard.

Reply:
Meaning: sometimes the flexibility is nice so you don't have to check the type before calling a function that's being strict about the type. I think this is generally good for all functions in this package.

Reply:
Looks like I just needed a couple type casts and it works out well even without the type guard.